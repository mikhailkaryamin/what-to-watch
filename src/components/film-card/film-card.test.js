import * as React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';

import FilmCard from './film-card.tsx';

describe(`Film card`, () => {
  const FILM_CARD = {
    backgroundImage: `img/aviator.jpg`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
    director: `Arthur Agee`,
    genre: `Comedy`,
    id: 1,
    isFavorite: false,
    name: `Devin Albert`,
    posterImage: `img/aviator.jpg`,
    previewImage: `img/aviator.jpg`,
    previewVideoLink: `link`,
    rating: 3,
    released: 2015,
    runTime: 100,
    scoreCount: 300,
    starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
    video: `some/link`,
  };

  const wrapper = renderer
    .create(
        <Router>
          <FilmCard
            film={FILM_CARD}
            isPlayVideo={false}
            onMouseEnter={() => {}}
            onMouseLeave={() => {}}
            onStopPreviewClick={() => {}}
          />
        </Router>
    ).toJSON();

  test(`should render film card`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});

