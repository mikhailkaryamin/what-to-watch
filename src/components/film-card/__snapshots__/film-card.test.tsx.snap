// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film card should render film card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Devin Albert"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Devin Albert
    </a>
  </h3>
</article>
`;
